File: jeanius/apitesting.py
import requests

# Replace with the actual URL of your API endpoint
API_ENDPOINT = "http://localhost:8000/api/measurements/"

# Replace with the path to the image file you want to test
IMAGE_PATH = "E:/study/FYP/fyp_dev/colorMatching/j1.jpg"

# Replace with the actual reference height value
REFERENCE_HEIGHT = 10.0

def test_measurement_api():
    try:
        files = {"image": open(IMAGE_PATH, "rb")}
        data = {"reference_height": str(REFERENCE_HEIGHT)}

        response = requests.post(API_ENDPOINT, files=files, data=data)

        if response.status_code == 200:
            results = response.json()
            print("API Response:")
            print(results)
        else:
            print("Error:", response.status_code)
            print(response.text)

    except Exception as e:
        print("Error:", str(e))


if __name__ == "__main__":
    test_measurement_api()

File: jeanius/apitesting.js
const axios = require('axios');
const FormData = require('form-data');
const fs = require('fs');

const url = "http://localhost:8000/api/color-match/";

// Create a FormData object
const formData = new FormData();

// Append files to the FormData object
formData.append('image1', fs.createReadStream('E:/study/FYP/fyp_dev/colorMatching/j1.jpg'), 'test1.jpg');
formData.append('image2', fs.createReadStream('E:/study/FYP/fyp_dev/colorMatching/j2.jpg'), 'test2.jpg');

// Send the request using axios
axios.post(url, formData, {
  headers: {
    ...formData.getHeaders(),
  },
})
  .then(response => {
    // Handle the response data
    console.log(response.status);
    console.log(response.data);
  })
  .catch(error => {
    // Handle errors
    console.error('Error:', error);
  });
